From 5efbcc10a3420e26859f8e915f0259a0103b7727 Mon Sep 17 00:00:00 2001
From: Adel Gadllah <adel.gadllah@gmail.com>
Date: Wed, 3 Jul 2013 18:51:53 +0200
Subject: [PATCH] offscreen: Allocate the framebuffer in
 cogl_offscreen_new_to_texture_full

The API says that it should return NULL on failure but it does not do that
due to the lazy allocation.

https://bugzilla.gnome.org/show_bug.cgi?id=703174
---
 cogl/cogl-framebuffer.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cogl/cogl-framebuffer.c b/cogl/cogl-framebuffer.c
index d9ec88d..9649e4d 100644
--- a/cogl/cogl-framebuffer.c
+++ b/cogl/cogl-framebuffer.c
@@ -618,6 +618,7 @@ _cogl_offscreen_new_to_texture_full (CoglTexture *texture,
   int level_width;
   int level_height;
   CoglOffscreen *ret;
+  CoglError *error = NULL;
 
   _COGL_RETURN_VAL_IF_FAIL (cogl_is_texture (texture), NULL);
   _COGL_RETURN_VAL_IF_FAIL (level < _cogl_texture_get_n_levels (texture),
@@ -649,6 +650,13 @@ _cogl_offscreen_new_to_texture_full (CoglTexture *texture,
 
   _cogl_texture_associate_framebuffer (texture, fb);
 
+  if (!cogl_framebuffer_allocate (ret, &error))
+    {
+      cogl_object_unref (offscreen);
+      cogl_error_free (error);
+      ret = NULL;
+    }
+
   return ret;
 }
 
-- 
1.8.3.1